### ISSUE TITLE
Installation is broken on a fresh install of npm/serverless (mac osx)

### ISSUE THREAD

**Issue by marksteele** [serverless/serverless-graphql-blog#11]
serverless project install serverless-graphql-blog

---

|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|**|  _**/|_____|**| |**|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, v0.5.1
`-------'

Serverless: Installing Serverless Project "serverless-graphql-blog"...
Serverless: Downloading project and installing dependencies...
/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:49
        fn = function () { throw arg; };
                           ^

ServerlessError: This plugin could not be found: serverless-optimizer-plugin
    at new ServerlessError (/usr/local/lib/node_modules/serverless/lib/Error.js:17:11)
    at Serverless._loadPlugins (/usr/local/lib/node_modules/serverless/lib/Serverless.js:293:17)
    at Serverless.loadProjectPlugins (/usr/local/lib/node_modules/serverless/lib/Serverless.js:240:10)
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:77:41
From previous event:
    at Serverless.init (/usr/local/lib/node_modules/serverless/lib/Serverless.js:73:25)
    at /usr/local/lib/node_modules/serverless/lib/actions/ProjectInstall.js:213:24
From previous event:
    at ProjectInstall._initProject (/usr/local/lib/node_modules/serverless/lib/actions/ProjectInstall.js:208:27)
From previous event:
    at ProjectInstall.installProject (/usr/local/lib/node_modules/serverless/lib/actions/ProjectInstall.js:110:12)
From previous event:
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:181:31
    at Array.reduce (native)
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:180:32
From previous event:
    at Serverless._execute (/usr/local/lib/node_modules/serverless/lib/Serverless.js:178:12)
    at Serverless.actions.(anonymous function) (/usr/local/lib/node_modules/serverless/lib/Serverless.js:424:20)
    at Serverless.command (/usr/local/lib/node_modules/serverless/lib/Serverless.js:393:38)
    at /usr/local/lib/node_modules/serverless/bin/serverless:19:16
    at processImmediate [as _immediateCallback](timers.js:383:17)
From previous event:
    at Object.<anonymous> (/usr/local/lib/node_modules/serverless/bin/serverless:18:4)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:141:18)
    at node.js:933:3

**Comment by Mcmartelle** [comment]
I'm getting a similar error

$ node -v
v4.4.2
$ npm -v
2.15.0
$ serverless project install serverless-graphql-blog

---

|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|**|  _**/|_____|**| |**|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, v0.5.1
`-------'

Serverless: Installing Serverless Project "serverless-graphql-blog"...  
Serverless: Downloading project and installing dependencies...  
/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:49
        fn = function () { throw arg; };
                           ^

TypeError: Class extends value #<Serverless> is not a function or null
    at module.exports.ServerlessPlugin.browserify._this.config.requires.map._this.config.plugins.map._this.config.transforms.map._this.config.exclude.forEach._this.config.ignore.forEach._generateIncludePaths (/Users/matthew.martelle/Desktop/personal/serverless-graphql-backend/serverless-graphql-blog/node_modules/serverless-optimizer-plugin/index.js:21:37)
    at Serverless._loadPlugins (/usr/local/lib/node_modules/serverless/lib/Serverless.js:286:25)
    at Serverless.loadProjectPlugins (/usr/local/lib/node_modules/serverless/lib/Serverless.js:240:10)
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:77:41
From previous event:
    at Serverless.init (/usr/local/lib/node_modules/serverless/lib/Serverless.js:73:25)
    at /usr/local/lib/node_modules/serverless/lib/actions/ProjectInstall.js:213:24
From previous event:
    at ProjectInstall._initProject (/usr/local/lib/node_modules/serverless/lib/actions/ProjectInstall.js:208:27)
From previous event:
    at ProjectInstall.installProject (/usr/local/lib/node_modules/serverless/lib/actions/ProjectInstall.js:110:12)
From previous event:
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:181:31
    at Array.reduce (native)
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:180:32
From previous event:
    at Serverless._execute (/usr/local/lib/node_modules/serverless/lib/Serverless.js:178:12)
    at Serverless.actions.(anonymous function) (/usr/local/lib/node_modules/serverless/lib/Serverless.js:424:20)
    at Serverless.command (/usr/local/lib/node_modules/serverless/lib/Serverless.js:393:38)
    at /usr/local/lib/node_modules/serverless/bin/serverless:19:16
    at processImmediate [as _immediateCallback](timers.js:383:17)
From previous event:
    at Object.<anonymous> (/usr/local/lib/node_modules/serverless/bin/serverless:18:4)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:139:18)
    at node.js:968:3

**Comment by ihistand** [comment]
This is reproducible on both Windows 10 and Ubuntu.

Any Ideas?

``serverless project install serverless-graphql-blog

---

|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|**|  _**/|_____|**| |**|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, v0.5.2
`-------'

Serverless: Installing Serverless Project "serverless-graphql-blog"...  
Serverless: Downloading project and installing dependencies...  
/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:49
        fn = function () { throw arg; };
                           ^

TypeError: Class extends value #<Serverless> is not a function or null
    at module.exports.ServerlessPlugin.browserify._this.config.requires.map._this.config.plugins.map._this.config.transforms.map._this.config.exclude.forEach._this.config.ignore.forEach._generateIncludePaths (/home/ivan/Projects/serverless-graphql-blog/node_modules/serverless-optimizer-plugin/index.js:21:37)
    at Serverless._loadPlugins (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:290:25)
    at Serverless.loadProjectPlugins (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:244:10)
    at /home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:75:41
From previous event:
    at Serverless.init (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:71:25)
    at /home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/actions/ProjectInstall.js:197:24
From previous event:
    at ProjectInstall._initProject (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/actions/ProjectInstall.js:192:27)
    at next (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/node_modules/rimraf/rimraf.js:74:7)
    at FSReqWrap.CB [as oncomplete](/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/node_modules/rimraf/rimraf.js:110:9)
From previous event:
    at ProjectInstall.installProject (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/actions/ProjectInstall.js:111:12)
From previous event:
    at /home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:185:31
    at Array.reduce (native)
    at /home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:184:32
From previous event:
    at Serverless._execute (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:182:12)
    at Serverless.actions.(anonymous function) (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:428:20)
    at Serverless.command (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/lib/Serverless.js:397:38)
    at /home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/bin/serverless:19:16
    at tryOnImmediate (timers.js:534:15)
    at processImmediate [as _immediateCallback](timers.js:514:5)
From previous event:
    at Object.<anonymous> (/home/ivan/.nvm/versions/node/v5.10.1/lib/node_modules/serverless/bin/serverless:18:4)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:146:18)
    at node.js:404:3

**Comment by sebastian-hsu** [comment]
same here (windows 7), any ideas?

$node -v
**v4.3.2**

$npm -v
**2.14.12**

---

|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|**|  _**/|_____|**| |**|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, **v0.5.5**
`-------'

**Comment by ValkyrieUK** [comment]
Same issue, potentially an issue with serverless-plugin-optimiser as an issue has been raised here https://github.com/serverless/serverless-optimizer-plugin/issues/23.

I followed the steps there and replaced the serverless optimiser line in the `package.json` with `"serverless-optimizer-plugin": "https://github.com/serverless/serverless-optimizer-plugin.git"`. After reinstalling npm modules npm errors and it deletes the current directory ! ![my shell](https://i.gyazo.com/51a625ad849bfc39ac5682f461914188.png)

**Comment by ryuheechul** [comment]
Same here

**Comment by ruid0** [comment]
Same for me 
node -v
v5.10.1
npm -v
3.8.3
![screen shot 2016-05-03 at 5 23 48 pm](https://cloud.githubusercontent.com/assets/2870909/14979265/dda15458-1153-11e6-8988-02cf91017ddb.png)

**Comment by DaveyEdwards** [comment]
Same here with 
Windows 10

node-v : 5.9.0
npm-v : 3.8.2

Is there no working work around for this?

**Comment by AlexMcConnell** [comment]
It works on Serverless 4.2. It breaks on Serverless 5+. Serverless is just updating so fast that it's difficult to keep up with.

**Comment by andyfen** [comment]
also

➜  serverless-graphql-blog serverless dash deploy
/Users/user/.nvm/versions/node/v5.2.0/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:61
        fn = function () { throw arg; };
                           ^

TypeError: Class extends value #<Serverless> is not a function or null
    at module.exports.ServerlessPlugin.browserify._this.config.requires.map._this.config.plugins.map._this.config.transforms.map._this.config.exclude.forEach._this.config.ignore.forEach._generateIncludePaths (/Users/user/Documents/serverless/serverless-graphql-blog/node_modules/serverless-optimizer-plugin/index.js:21:37)
    at Serverless._loadPlugins (/Users/user/.nvm/versions/node/v5.2.0/lib/node_modules/serverless/lib/Serverless.js:291:25)
    at Serverless.loadProjectPlugins (/Users/user/.nvm/versions/node/v5.2.0/lib/node_modules/serverless/lib/Serverless.js:245:10)
    at /Users/user/.nvm/versions/node/v5.2.0/lib/node_modules/serverless/lib/Serverless.js:76:41
From previous event:
    at Serverless.init (/Users/user/.nvm/versions/node/v5.2.0/lib/node_modules/serverless/lib/Serverless.js:72:25)
    at Object.<anonymous> (/Users/user/.nvm/versions/node/v5.2.0/lib/node_modules/serverless/bin/serverless:17:12)
    at Module._compile (module.js:399:26)
    at Object.Module._extensions..js (module.js:406:10)
    at Module.load (module.js:345:32)
    at Function.Module._load (module.js:302:12)
    at Function.Module.runMain (module.js:431:10)
    at startup (node.js:141:18)
    at node.js:977:3

**Comment by alibasta** [comment]
same here:

Alexanders-Air:Serverless alibasta$ node --version
v6.2.0
Alexanders-Air:Serverless alibasta$ npm --version
3.8.9

**Comment by waltervargas** [comment]
Same for me: 

```
$ node -v                                                                                                                    
v5.10.1
$ npm -v
3.8.3
$ uname -a 
Darwin wrkstation-pro.local 15.5.0 Darwin Kernel Version 15.5.0: Fri Apr 22 18:59:34 PDT 2016; root:xnu-3248.50.21~19/RELEASE_X86_64 x86_64
```

**Comment by kevinold** [comment]
Hi,

I've finally had a chance to update the structure of this project for the 0.5 structure.

I've published the update as version 0.0.2 on npm.

https://github.com/serverless/serverless-graphql-blog/commit/7824422a4741097765991cceab8f78328606911b

Closing this issue for now.  Please open a separate issue if you run into other issues.

**Comment by wilsonmar** [comment]
I think this was closed prematurely. I installed from node download today to get v4.4.5 and npm 2.15.5, serverless 0.5.6. Trouble came near the end of : serverless project install serverless-starte.

Serverless: Creating region "us-west-2" in stage "dev"...  
Serverless: Deploying resources to stage "dev" in region "us-west-2" via Cloudformation (~3 minutes)...  
/Users/mac/.npm-packages/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:61
        fn = function () { throw arg; };
